packages/shared/src/graphql-server/types/gql.ts

```diff
@@ -111,6 +113,7 @@ export const typeDefs = gql`
     myMatches(anonymousUserId: String = null, offset: Int! = 0, count: Int! = 50): CombatQueryResult!
     userMatches(userId: String!, offset: Int! = 0, count: Int! = 50): CombatQueryResult!
     matchesWithCombatant(playerName: String!): [CombatDataStub!]!
+    matchById(matchId: String!, anon: Boolean!): CombatDataStub!
```

Comment:
we have lots of places that need to understand and keep track of anon vs not. I wonder how many of these can be eliminated and kept dumb.. maybe not for this initial pass but feels like it should be possible to reduce complexity here. thoughts?

Reply:
Yeah this pass starts that process. There's a few different routes we could go to scope the anon knowledge out of consumer code but I still need to review what consequences each will have